app/src/main/java/com/infomaniak/mail/data/models/thread/Thread.kt

```diff
@@ -15,71 +15,92 @@
-
 package com.infomaniak.mail.data.models.thread
 
 import android.content.Context
 import androidx.annotation.IdRes
 import com.infomaniak.lib.core.utils.*
 import com.infomaniak.mail.R
-import com.infomaniak.mail.data.api.RealmInstantSerializer
-import com.infomaniak.mail.data.api.RealmListSerializer
 import com.infomaniak.mail.data.models.correspondent.Recipient
 import com.infomaniak.mail.data.models.message.Message
 import com.infomaniak.mail.utils.isSmallerThanDays
 import com.infomaniak.mail.utils.toDate
 import io.realm.kotlin.ext.realmListOf
+import io.realm.kotlin.ext.realmSetOf
+import io.realm.kotlin.ext.toRealmList
+import io.realm.kotlin.ext.toRealmSet
 import io.realm.kotlin.types.RealmInstant
 import io.realm.kotlin.types.RealmList
 import io.realm.kotlin.types.RealmObject
+import io.realm.kotlin.types.RealmSet
 import io.realm.kotlin.types.annotations.PrimaryKey
-import kotlinx.serialization.SerialName
-import kotlinx.serialization.Serializable
-import kotlinx.serialization.Transient
-import kotlinx.serialization.UseSerializers
 
-@Serializable
 class Thread : RealmObject {
 
-    //region API data
     @PrimaryKey
     var uid: String = ""
-    @SerialName("messages_count")
-    var messagesCount: Int = 0
-    @SerialName("unique_messages_count")
-    var uniqueMessagesCount: Int = 0
-    @SerialName("deleted_messages_count")
-    var deletedMessagesCount: Int = 0
     var messages: RealmList<Message> = realmListOf()
-    @SerialName("unseen_messages")
+    var uniqueMessagesCount: Int = 0
     var unseenMessagesCount: Int = 0
     var from: RealmList<Recipient> = realmListOf()
-    var cc: RealmList<Recipient> = realmListOf()
-    var bcc: RealmList<Recipient> = realmListOf()
     var to: RealmList<Recipient> = realmListOf()
     var subject: String? = null
     var date: RealmInstant = RealmInstant.MAX
-    @SerialName("has_attachments")
+    var size: Int = 0
     var hasAttachments: Boolean = false
-    @SerialName("has_st_attachments")
-    var hasStAttachments: Boolean = false
-    @SerialName("has_drafts")
     var hasDrafts: Boolean = false
-    @SerialName("flagged")
     var isFavorite: Boolean = false
     var answered: Boolean = false
     var forwarded: Boolean = false
     var scheduled: Boolean = false
-    var size: Int = 0
-    //endregion
+    var foldersIds: RealmSet<String> = realmSetOf()
+    var messagesIds: RealmSet<String> = realmSetOf()
+
+    fun addMessage(message: Message) {
+        message.threadUid = uid
+        messages.add(message)
+        messagesIds = messages.flatMap { it.messageIds }.toRealmSet()
+    }
+
+    fun removeMessage(message: Message) {
+        messages.removeIf { it.uid == message.uid }
+        recomputeThread()
+    }
 
-    //region Local data (Transient)
-    @Transient
-    var mailboxUuid: String = ""
-    @Transient
-    var folderId: String = ""
-    //endregion
+    fun recomputeThread() {
+
+        messages.sortBy { it.date }
```

Comment:
Because the Messages are not added chronologically into the Thread, but only when a Folder is open.
If we first open a Folder with recent Messages, then we open a Folder with older Messages, the older will be added after the newest.
So when adding Messages to a Thread, we need to order them again.